sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/CollationKey.scala

```diff
@@ -0,0 +1,161 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.expressions
+
+import java.util.Locale
```

Comment:
Let's avoid using JVM's locale. We will anyhow probably completely switch to ICU even for UTF8_BINARY_LCASE.

If I am reading code correctly you don't even need Locale here. Just use `str.toLowerCase` instead of `str.toString.toLowerCase(Locale.ROOT)`

Reply:
agreed, will change this